source/extensions/filters/network/mongo_proxy/bson_impl.cc

```diff
@@ -101,6 +101,11 @@ std::string BufferHelper::removeString(Buffer::Instance& data) {
   }
 
   char* start = reinterpret_cast<char*>(data.linearize(length));
+  if (start[length - 1]) {
```

Comment:
Just eliminate the conditional. Drop lines 104, and 109-111.

Reply:
Done.